### ISSUE TITLE
[Google Search API] Searches return inconsistent number of `organic_results` with the `num` parameter

### ISSUE THREAD

**Issue by marm123** [serpapi/public-roadmap#1061]
One of our customers reported that Google returns less than ten results even though the `num=10` (and even though ten is the default value). I'm not sure if there is anything we can do about that, but maybe we can look into it and see if we can make the `num` parameter for mobile more consistent.

Search with `num=10` and 9 `organic_results`:
<img width="1436" alt="image" src="https://github.com/serpapi/public-roadmap/assets/28161162/9fe40e8d-ddb4-45fd-bdcc-441c934ab011">

Search with `num=10` and 6 `organic_results`:
<img width="1438" alt="image" src="https://github.com/serpapi/public-roadmap/assets/28161162/163e5b14-0120-4fb6-99aa-d81810dede8c">

[Playground - 9 results](https://serpapi.com/playground?q=Grapes&location=Austin%2C+Texas%2C+United+States&gl=us&hl=en&num=10&device=mobile) | [Inspect - 9 results](https://serpapi.com/searches/64e488162fe30255388d8ae9/inspect)
[Playground - 6 results](https://serpapi.com/playground?q=can+guinea+pigs+eat+grapes&location=Austin%2C+Texas%2C+United+States&gl=us&hl=en&num=10&device=mobile) | [Inspect - 6 results](https://serpapi.com/searches/64e48834c999039bdfd55be9/inspect)
[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000047235?view=List)

**Comment by martin-serpapi** [comment]
Another customer reported this. In their case it is happening on Desktop.

Search with `num=1` and 0 `organic_results`:

![image](https://github.com/serpapi/public-roadmap/assets/144151895/dc4a7884-4abc-4380-9690-b53fb74f05eb)

[Playground - num=1 & 0 organic_results](https://serpapi.com/playground?q=Cute+Cat%3F&gl=us&hl=en&safe=active&start=0&num=1) | [Inspect num=1 & 0 organic_results](https://serpapi.com/searches/d17ed9d9e43ff379/65545d03e4a23d4ee903c02b.json)

I was able to reproduce it with `q=Coffee` & `num=10` (`safe=off` & `filter=0` are set as well):

![image](https://github.com/serpapi/public-roadmap/assets/144151895/594aa02f-7499-4bcf-bb4a-f17c3e8192a8)

[Playground - num=10 & 8 organic_results](https://serpapi.com/playground?q=Coffee&location=Denver%2C+Colorado%2C+United+States&gl=us&hl=en&safe=off&filter=0&start=0&num=10) | [Inspect - num=10 & 8 organic_results](https://serpapi.com/searches/865d121cc2019281/6554934dc504e9b53036f680.json)

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000053783?view=List)

**Comment by martin-serpapi** [comment]
Adding `urgent` label as this is a crucial regression in our main API.

**Comment by marm123** [comment]
Another customer is having this issue. I'll share Inspect pages once provided.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/view/99173/conversation/102910000053478?view=List)

**Comment by aliayar** [comment]
I am not sure if this is an issue on our end. The default Google behavior is returning 10 results. Adding `num=10` might be interpreted by Google as an indication of scraping activity hence the issues.

Martin's latest report even includes more of a bizarre situation where user tries to `num=1`. Google doesn't provide a UI for setting up `num` parameter on their results page. Due to that, only technical users are supposed to use `num` parameter by manually updating the url to add the `num` parameter. 

Since Google also recently changed their pagination with endless scrolling, I believe that adding `start=0` or adding `num` parameter with a number other than 10 or doubles of 10 creates issues for users.

Those who follow the defaults of Google search engine has never reported such issues and only reports that we have received are from users who mess with defaults parameters in unexpected ways.

I will change the label back to `ready` until one of our engineers investigates this and makes an informed decision regarding the future of this report.

**Comment by martin-serpapi** [comment]
I've tested sending a request without `num` and `start` parameters and with `safe=off` and with `filter=0`. There are still only 8 `organic_results` returned:

![image](https://github.com/serpapi/public-roadmap/assets/144151895/fa214b7e-2216-441e-a0b6-f28f0867d1ab)

[Playground](https://serpapi.com/playground?q=Coffee&location=New+York%2C+NY%2C+United+States&gl=us&hl=en&safe=off&filter=0) | [Inspect](https://serpapi.com/searches/6554c0711988e5ed286f5f71/inspect)

I've executed the same request in Google, connected to a New York VPN, and Google is returning only 8 organic results in the response as well:

![image](https://github.com/serpapi/public-roadmap/assets/144151895/ea4bc6a2-0941-4319-9046-890e58bb3dc6)

<img width="1430" alt="image" src="https://github.com/serpapi/public-roadmap/assets/144151895/49658787-5557-43e6-8485-2103620f69f4">

[Google Search](https://www.google.com/search?q=coffee&oq=coffee&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIGCAEQRRg7MgYIAhBFGDsyBggDEEUYOzIHCAQQABiPAjIGCAUQRRg8MgYIBhBFGDwyBggHEEUYPNIBCDI0MTBqMGoxqAIAsAIA&sourceid=chrome&ie=UTF-8&filter=0#ip=1)

I've also tested sending multiple requests to our Google Search API using a Node.js script for a few different search queries, using the following parameters:

```javascript
const response = await getJson({
            engine: "google",
            q: topic,
            location: "New York, NY, United States",
            google_domain: "google.com",
            hl: "en",
            gl: "us",
            filter: 0,
            safe: "off",
            async: true,
            api_key: API_KEY
});
```

There are different number of `organic_results` returned for the different queries. 

<img width="1084" alt="image" src="https://github.com/serpapi/public-roadmap/assets/144151895/9d4b5cec-24ca-4240-8235-692a9bcdb960">

This behaviour seems to be coming from Google itself.

Shouldn't we restrict using a `num` parameter with a value less than 10, or a value that is not an increment of 10, on our end, if it's not supposed be used with other values? The `start` parameter as well, that 0 cannot be passed as a value, and only 10 and increments of 10 can be passed.

**Comment by hartator** [comment]
Yes, Google returns a number of organic results that are variable. `num` is not a guarantee.

I think it can be interesting to expand on @martin-serpapi's research and make a blog post just about `num` behaviors, expected values, and history.

> Shouldn't we restrict using a `num` parameter with a value less than 10, or a value that is not an increment of 10, on our end, if it's not supposed be used with other values? The `start` parameter as well, that 0 cannot be passed as a value, and only 10 and increments of 10 can be passed.

I think we already talked about it [somewhere](https://github.com/search?q=repo%3Aserpapi%2FSerpApi+num). I don't remember what was our reasoning to not restrict `num`. I think some customers had valid use case with more creative use of `num`. E.g., `num=1` to just have results smaller and faster.

Closing this as non-fixable.
